### ISSUE TITLE
gl-react: a new special 'symbol' to pass to uniforms to inject a texture size as vec2

### ISSUE THREAD

**Issue by gre** [gre/gl-react#116]
> you don't want to solve this twice.. because you had either to know the size in advance or to load it yourself, on user-side.. also, it's getting hard for when introducing #115 (uniforms needs to be consistent when the texture is sent).

## API

I think something like

```
textureSize(input)
```

not sure about the naming, but basically this needs to return a special object we will internally recognize. textureSize() returned value is a black box for user and is meant to only be given to a Node#uniforms and will be sent to the shader as `uniform vec2`.
internally, we replace it by the size (in vec2) when we know it, or we send just [0.0, 0.0] until then.

important to note that input it the same "descriptor" you sent for the texture.

for instance
```js
uniforms={{
  image: "http://i.imgur.com/26UxJ2g.png",
  size: textureSize("http://i.imgur.com/26UxJ2g.png")
}}
```

or less trivial example:

```js
uniforms={{
  image: ndArray,
  size: textureSize(ndArray)
}}
```

also not sure what to think about this one:

```js
uniforms={{
  image: () => this.refs.canvas,
  size: textureSize(() => this.refs.canvas)
}}
```

you can obviously factorize in a var ;)

and also there will be edgecase like vdom. we will have to make sure there is no edge case we don't cover (and make sure the error is clear).

hopefully for things like composition (send a react element to a uniform) we can figure out the mapping because at react render() time we should have referencial transparency on these!

## Impl

so basically i think it could be a new method on a Loader.. like a getSize(). it would only get called after a load() or a successful get().
there will be specific case for the HTMLElement as input, but it's solvable (as i remember, .videoWidth,.videoHeight for video will be an important bit, otherwise they all work with .width,.height)

**Comment by gre** [comment]
note that the usecase is not necessarily to use the width & height in the shader, but usually the ratio `width/height` is interesting value to write code that preserve it. and don't stretch everything.

might be useful for impl things like https://github.com/gre/gl-react-image

**Comment by gre** [comment]
probably going to make a symbol to get the ratio, not the actual size. that way we can default it to 1 when no available which will mean treating things like currently, stretched

**Comment by gre** [comment]
so what i'll make is that when the `load()` succeed, is it assumed it also loaded the dimension, meaning that you can `getRatio(input)` it
